container/go/cmd/puller/puller.go

```diff
@@ -41,67 +43,64 @@ var (
 // image was (probably) never tagged with this, but lets us avoid applying the
 // ":latest" tag which might be misleading.
 const iWasADigestTag = "i-was-a-digest"
 
-// NOTE: This function is mostly copied from https://github.com/google/go-containerregistry/blob/master/pkg/crane/pull.go
+// NOTE: This function is adapted from https://github.com/google/go-containerregistry/blob/master/pkg/crane/pull.go
 // with slight modification to take in a platform argument.
 // Pull the image with given <imgName> to destination <dstPath> with optional
 // cache files and required platform specifications.
-func pull(imgName, dstPath, cachePath string, platform v1.Platform) {
-	// Get a digest/tag based on the name
+func pull(imgName, dstPath string, platform v1.Platform) {
+	// Get a digest/tag based on the name.
 	ref, err := name.ParseReference(imgName)
 	if err != nil {
 		log.Fatalf("parsing tag %q: %v", imgName, err)
 	}
 	log.Printf("Pulling %v", ref)
 
-	// Fetch the image with desired cache files and platform specs
-	i, err := remote.Image(ref, remote.WithAuthFromKeychain(authn.DefaultKeychain), remote.WithPlatform(platform))
+	// Fetch the image with desired cache files and platform specs.
+	img, err := remote.Image(ref, remote.WithAuthFromKeychain(authn.DefaultKeychain), remote.WithPlatform(platform))
 	if err != nil {
 		log.Fatalf("reading image %q: %v", ref, err)
 	}
 
-	// WriteToFile wants a tag to write to the tarball, but we might have
-	// been given a digest.
-	// If the original ref was a tag, use that. Otherwise, if it was a
-	// digest, tag the image with :i-was-a-digest instead.
-	tag, ok := ref.(name.Tag)
-	if !ok {
-		d, ok := ref.(name.Digest)
-		if !ok {
-			log.Fatal("ref wasn't a tag or digest")
-		}
-		s := fmt.Sprintf("%s:%s", d.Repository.Name(), iWasADigestTag)
-		tag, err = name.NewTag(s)
-		if err != nil {
-			log.Fatalf("parsing digest as tag (%s): %v", s, err)
-		}
+	// // Image file to write to disk.
+	if err := oci.Write(img, dstPath); err != nil {
+		// if err := oci.Write(img, path); err != nil {
+		log.Fatalf("failed to write image to %q: %v", dstPath, err)
 	}
+}
+
+func main() {
+	flag.Parse()
+	log.Println("Running the Image Puller to pull images from a Docker Registry...")
 
-	if err := tarball.WriteToFile(dstPath, tag, i); err != nil {
-		log.Fatalf("writing image %q: %v", dstPath, err)
+	if *imgName == "" {
+		log.Fatalln("Required option -name was not specified.")
 	}
+	if *directory == "" {
+		log.Fatalln("Required option -directory was not specified.")
+	}
+
+	// If the user provided a client config directory, instruct the keychain resolver
+	// to use it to look for the docker client config
+	if *clientConfigDir != "" {
+		ospkg.Setenv("DOCKER_CONFIG", *clientConfigDir)
+	}
+
+	// Create a Platform struct with arguments
+	platform := v1.Platform{
+		Architecture: *arch,
+		OS:           *os,
+		OSVersion:    *osVersion,
+		OSFeatures:   strings.Fields(*osFeatures),
```

Comment:
strings.Fields assumes the input strings are separated by space. Specify this explicitly in the flag documentation for osFeatures and features.